app/lib/email/send-magic-link-email.js

```diff
@@ -1,13 +1,21 @@
 const { v4: uuid } = require('uuid')
 const sendEmail = require('./send-email')
-const { serviceUri } = require('../../config')
+const { serviceUri, testToken } = require('../../config')
 const { notify: { templateIdFarmerApplyLogin, templateIdFarmerClaimLogin, templateIdVetLogin } } = require('../../config')
 const { farmerApply, farmerClaim, vet } = require('../../config/user-types')
-
+const { getByEmail } = require('../../api-requests/users')
+async function getToken (email) {
+  const user = await getByEmail(email);
```

Comment:
Rather than retrieving the user first, it might be worth checking if there is a token which would then mean the user can be retrieved. Doing this would reduce the calls to the `getByEmail` function.

Reply:
Done